File: app/src/main/java/com/muhsanjaved/databinding_practice/viewModels/MainViewModel.java
package com.muhsanjaved.databinding_practice.viewModels;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;

import com.muhsanjaved.databinding_practice.models.User;

public class MainViewModel extends AndroidViewModel {

    private User mUser;

    public MainViewModel(@NonNull Application application) {
        super(application);
        mUser = new User("Ali", 20);
    }

    public User getUser(){
        return this.mUser;
    }
}

File: app/src/main/java/com/muhsanjaved/databinding_practice/viewModels/MainViewModel.kt
package com.muhsanjaved.databinding_practice.viewModels

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import com.muhsanjaved.databinding_practice.models.DataProvider
import com.muhsanjaved.databinding_practice.models.Product

class MainViewModel(application: Application) : AndroidViewModel(application) {

    private var product = DataProvider.productList.get(1)

    fun getProduct(): Product = product

}